lib/charms/operator_libs_linux/v0/grub.py

```diff
@@ -161,13 +161,24 @@ def _save_config(path: Path, config: Dict[str, str], header: str = CONFIG_HEADER
     if path.exists():
         logger.debug("GRUB config %s already exist and it will overwritten", path)
 
-    context = [f"{key}={shlex.quote(value)}" for key, value in config.items()]
     with open(path, "w", encoding="UTF-8") as file:
-        file.writelines([header, *context])
+        file.write(f"{header}{os.linesep}")
```

Comment:
There's no need to use `os.linesep` here, because we know we're on Linux / POSIX, so `\n` is simpler and clearer. In addition, as the [docs for `os.linesep`](https://docs.python.org/3/library/os.html#os.linesep) say, "Do not use os.linesep as a line terminator when writing files opened in text mode (the default); use a single '\n' instead, on all platforms."

```suggestion
        file.write(header + "\n")
```

Or maybe the header string could just include the `\n` in it?

Reply:
Actually header is providing `\n`, so it's not needed. Regarding `os.linesep`, I read the docs and I must say thanks for pointing this out. I'm always saying everyone to use it everywhere, it's it's not desired at all. I'll replace it with `\n`.